### DISCUSSION TITLE
The result of minimizing a comment behaves differently via GraphQL & GitHub UI button

### DISCUSSION THREAD

**Discussion by plainheart** [community/community#19865] [Projects and Issues]
Hi, I'm using the GraphQL [`minimizeComment`](https://docs.github.com/en/graphql/reference/mutations#minimizecomment) mutation to hide the outdated or resolved comments. But it seems the GraphQL mutation doesn't behave the same as the GitHub UI button. See the pictures below,

| GraphQL | UI button |
| :----: | :----: |
| ![image](https://user-images.githubusercontent.com/26999792/176985294-de3c3e89-31df-4d61-8638-e574317942c7.png) | ![image](https://user-images.githubusercontent.com/26999792/176985299-8c15f142-2252-4521-ac8d-efed1e58b7cb.png) |

This is the request content and parameter

```graphql
mutation minimizeComment($id: ID!, $classifier: ReportedContentClassifiers!) {
    minimizeComment(input: { subjectId: $id, classifier: $classifier }) {
        clientMutationId
        minimizedComment {
            isMinimized
            minimizedReason
            viewerCanMinimize
        }
    }
}
```

The parameters are,

```js
{
  "id": comment.node_id,
  "classifier": 'OUTDATED'
}
```

This is the response from GraphQL. It indicated that the comment was minimized with the expected reason. But the GitHub UI only shows the information like `This comment has been minimized.`, with no specific reason.

```json
{
  "minimizeComment": {
    "clientMutationId": null,
    "minimizedComment": {
      "isMinimized": true,
      "minimizedReason": "outdated",
      "viewerCanMinimize": true
    }
  }
} 
```

Is this different behavior by design?

**Comment by plainheart** [comment]
Hi, can someone help look into this issue? Thanks!

**Comment by willfish** [comment]
If you fetch these comments via the graphql query api you can see that the enum is setting the wrong value:

```jso
[
  {
    "id": "<redacted>",
    "body": "**Terraform plan output for development:**\n```\n\nNo changes. Your infrastructure matches the configuration.\n\nTerraform has compared your real infrastructure against your configuration\nand found no differences, so no changes are needed.",
    "isMinimized": true,
    "minimizedReason": "outdated"
  },
  {
    "id": "<redacted>",
    "body": "Testing with a comment",
    "isMinimized": true,
    "minimizedReason": "OUTDATED"
  }
]
```

**Reply by willfish** [reply]
The last comment, here is OUTDATED, I suspect we're seeing a fallback when they read the value out of the database

**Comment by SleipRecx** [comment]
I'm also experiencing this behaviour, someone looking into this?

**Comment by joshdk** [comment]
I ran into this behavior today, which is certainly a bug in the [minimizeComment GraphQL API](https://docs.github.com/en/graphql/reference/mutations#minimizecomment).

The comment that @willfish left was very insightful in understanding what is going on. It appears that the `classifier` enum is being **lowercased** internally **after** GitHub processes the API request. This now-invalid enum value results in the fallback "This comment has been minimized" message being displayed instead of the specific reason.

I was able to repro and verify this behavior using some Go code, but nowhere is it more obvious then using the GraphQL Explorer itself:

![image](https://github.com/community/community/assets/307183/faced4cb-8cf6-44fd-ab8b-15618ff2a5d6)

As an additional debugging point, the github.com web UI uses a completely separate (and inaccessible/undocumented) API to minimize comments (which correctly preserves the uppercase `classifier`), so this isn't a simple issue of e.g. incorrect GraphQL query inputs...

---

It would be very nice to have this bug fixed, as is is currently not possible to programmatically minimize comments for **specific** reasons. 

Thank you!

**Comment by kylemclaren** [comment]
Recently ran into this and can confirm that the issue is exactly as @joshdk described. Would be nice if someone from GH could take a look :)

**Comment by josephwynn-sc** [comment]
Can confirm, if I try to use the lowercase `minimizedReason` the API returns `Expected "outdated" to be one of: SPAM, ABUSE, OFF_TOPIC, OUTDATED, DUPLICATE, RESOLVED`. However these uppercase reasons seem to be ignored by the UI.

**Comment by reduckted** [comment]
How do we get someone from GitHub to actually look at this after three and a half years?

**Reply by TBeijen** [reply]
Amazing how, with all the schemas and contracts we have these days, this type of bugs pop up. 

This gives me strong Deja-vu with: https://github.com/Azure/azure-sdk-for-go/issues/23446 although almost certainly unrelated. I suppose not  being picky about casing is a MS thing (cough IE5 html, cough filesystem)

Kidding aside, hope GH can fix this. It just greatly helps to show the intent of comments being programmatically hidden.
